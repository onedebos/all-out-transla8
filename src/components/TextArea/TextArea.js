import styled from 'styled-components';

export default function TextArea({ isTarget, translation }) {
	return (
		<TextAreaWrapper
			isTarget={isTarget}
			rows="4"
			cols="50"
			disabled={isTarget ? true : false}
			placeholder={!isTarget ? 'Start Typing...' : ''}
		>
			{translation ? translation : ''}
		</TextAreaWrapper>
	);
}

const TextAreaWrapper = styled.textarea`
	margin-top: 20px;
	border-radius: 8px;
	width: 100%;
	max-width: 330px;
	margin-right: 10px;
	padding: 10px;
	font-size: 24px;
	background-color: ${(props) => (props.isTarget ? '#f8f9fa' : '')};
`;
