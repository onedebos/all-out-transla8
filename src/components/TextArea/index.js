import styled from 'styled-components';
import constants from '../../container/LanguagesSelector/constants'

export default function TextArea({ isTarget, translation, onChange }) {

	const {SRC_LANG, TRGT_LANG} = constants

	return (
		<TextAreaWrapper
			aria-label="text"
			isTarget={isTarget}
			rows="4"
			cols="50"
			disabled={isTarget ? true : false}
			placeholder={!isTarget ? 'Start Typing...' : ''}
			defaultValue={translation ? translation : ''}
			onChange={onChange && onChange}
			data-testid={isTarget ? TRGT_LANG + '-textarea' : SRC_LANG + '-textarea'}
		/>
	);
}

const TextAreaWrapper = styled.textarea`
	margin-top: 20px;
	border-radius: 8px;
	width: 100%;
	max-width: 330px;
	margin-right: 10px;
	padding: 10px;
	font-size: 24px;
	background-color: ${(props) => (props.isTarget ? '#f8f9fa' : '')};
`;
