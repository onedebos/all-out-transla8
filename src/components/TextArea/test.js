/**
 * @jest-environment jsdom
 */
 import { render, screen, cleanup } from '@testing-library/react';
 import '@testing-library/jest-dom';
 import renderer from 'react-test-renderer';
 import TextArea from '.';
 
 afterEach(() => {
     cleanup();
 });
 
 describe('<TextArea />', () => {

     it('renders the Button component successfully without crashing', () => {
         render(<TextArea isTarget={false} />);
     });

    
     it('matches snapshot', () => {
         const tree = renderer.create(<TextArea />).toJSON();
         expect(tree).toMatchSnapshot();
     });
 });
 