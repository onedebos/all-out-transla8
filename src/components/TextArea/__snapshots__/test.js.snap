// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextArea /> matches snapshot 1`] = `
<textarea
  aria-label="text"
  className="TextArea__TextAreaWrapper-sc-18mg7qw-0 cA-dkzx"
  cols="50"
  data-testid="SourceLanguage-textarea"
  defaultValue=""
  disabled={false}
  placeholder="Start Typing..."
  rows="4"
/>
`;
